scipy/stats/_fit.py

```diff
@@ -565,3 +566,528 @@ def nllf(free_params, data=data):  # bind data NOW
+    Carlo samples, and for most distributions in SciPy, distribution fitting
+    performed via numerical optimization. For this reason, it may be tempting
+    to treat parameters of the distribution pre-fit to `data` (by the user)
+    as though they were `known_params`, as specification of all parameters of
+    the distribution precludes the need to fit the distribution to each Monte
+    Carlo sample. (This is essentially how the original Kilmogorov-Smirnov
+    test is performed.) Although such a test can provide evidence against the
+    null hypothesis, the power of the test is low (that is, it is less likely
+    to reject the null hypothesis even when the null hypoothesis is false)
+    because the Monte Carlo samples are less likely to agree with the
+    null-hypothesized distribution as well as `data`. This tends to increase
+    the values of the statistic recorded in the null distribution, so that a
+    larger number of them exceed the value of statistic for `data`, thereby
+    inflating the p-value.
+
+    References
+    ----------
+    .. [1] M. A. Stephens (1974). "EDF Statistics for Goodness of Fit and
+           Some Comparisons." Journal of the American Statistical Association,
+           Vol. 69, pp. 730-737.
+    .. [2] B. Phipson and G. K. Smyth (2010). "Permutation P-values Should
+           Never Be Zero: Calculating Exact P-values When Permutations Are
+           Randomly Drawn." Statistical Applications in Genetics and Molecular
+           Biology 9.1.
+    .. [3] H. W. Lilliefors (1967). "On the Kolmogorov-Smirnov test for
+           normality with mean and variance unknown." Journal of the American
+           statistical Association 62.318: 399-402.
+
+    Examples
+    --------
+    A well-known test of the null hypothesis that data were drawn from a
+    given distribution is the Kolmogorov-Smirnov test, available in SciPy
+    as `scipy.stats.ks_1samp`. Suppose we wish to test whether the following
+    data:
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> rng = np.random.default_rng()
+    >>> x = stats.uniform.rvs(size=75, random_state=rng)
+
+    were sampled from a normal distribution. To perform a KS test, the
+    empirical distribution function of the observed data will be compared
+    against the (theoretical) cumulative distribution function of a normal
+    distribution. Of course, to do this, the normal distribution under the null
+    hypothesis must be fully specified. This is commonly done by first fitting
+    the ``loc`` and ``scale`` parameters of the distribution to the observed
+    data, then performing the test.
+
+    >>> loc, scale = np.mean(x), np.std(x, ddof=1)
+    >>> cdf = stats.norm(loc, scale).cdf
+    >>> stats.ks_1samp(x, cdf)
+    KstestResult(statistic=0.1119257570456813, pvalue=0.2827756409939257)
+
+    An advantage of the KS-test is that the p-value - the probability of
```

Comment:
I think we should point out more clearly that this is bad practice since it is a combined hypothesis. let me try to find a reference

Reply:
I have a reference that basically says it is nice for historical reasons but should never be used anymore. I'll add that .
I do think we show here why it's not great: it's really conservative. That is, using a KS test when parameters are not really known can _only increase_ the p-value. The statistic value of the observed data will be the same, but the rest of the null distribution will tend toward higher values than it would in a test that does not assume the parameters are known.

Reply:
Cool, it would be nice if you add a reference

Reply:
> I have a reference that basically says it is nice for historical reasons but should never be used anymore. 

Unfortunately, it (D'Agostino and Stephens ["Goodness of fit Techniques"](https://www.hep.uniovi.es/sscruz/Goodness-of-Fit-Techniques.pdf), page 406) qualifies the statement with "For testing for normality,..." 

![image](https://user-images.githubusercontent.com/6570539/192425962-fff14dbf-529e-4f57-8c16-f3f887943378.png)

So it not damning of the KS test in general. 

> Cool, it would be nice if you add a reference

I haven't seen a reference that makes statements as broad as the ones I make in the documentation. The statement you found in the Wikipedia page for [Lilliefors test](https://en.wikipedia.org/wiki/Lilliefors_test) is

> Thus the "null distribution" of the test statistic, i.e. its [probability distribution](https://en.wikipedia.org/wiki/Probability_distribution) assuming the null hypothesis is true, is [stochastically smaller](https://en.wikipedia.org/wiki/Stochastic_order) than the Kolmogorov–Smirnov distribution. This is the Lilliefors distribution.

 The statement that the null distribution of Lilliefors is "stochastically smaller" than that of KS agrees with what I wrote about the null distribution when we assume that parameters are known:

> This tends to increase the values of the statistic recorded in the null distribution

In the context of the KS statistic and the normal distribution, I was saying that the KS null distribution is "stochastically greater" than the Lilliefors distribution.  But my statement is more general because I didn't limit it to context of the KS statistic and the normal distribution.